Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-categories/ContestCategories.tsx

```diff
@@ -60,7 +77,57 @@ const ContestCategories = ({
         [ defaultSelected, categoriesFlat, getParents ],
     );
 
-    const handleTreeLabelClick = useCallback((node: ITreeItemType) => {
+    const strategyFilterGroup = useMemo(
+        () => possibleFilters.filter(({ type }) => type === FilterType.Strategy),
+        [ possibleFilters ],
+    );
+
+    const addNewStrategyFilters = useCallback(
+        (id: string, { allowedStrategyTypes }: ICategoryStrategiesTypes) => {
+            const strategyFiltersToAdd = allowedStrategyTypes
+                ?.map((value) => value.id.toString())
+                .map((x) => strategyFilterGroup.find((y) => y.value === x))
+                .filter((x) => !isNil(x)) as IFilter[];
+
+            setFilterProps([
+                ...filterProps,
+                { categoryId: id, strategies: strategyFiltersToAdd },
+            ]);
+        },
+        [ filterProps, strategyFilterGroup ],
+    );
+
+    const removeOldStrategyFilters = useCallback(
+        (id: string) => {
+            const newFilterProps = filterProps.filter((fp) => fp.categoryId !== id);
+
+            setFilterProps(newFilterProps);
+        },
+        [ filterProps ],
+    );
+
+    useEffect(
+        () => {
+            const flatteredStrategyFilters = filterProps.flatMap((s) => s.strategies);
+
+            setStrategyFilters(uniq(flatteredStrategyFilters));
+            setCurrentCurrentStrategyFilters(filterProps);
+        },
+        [ filterProps, setStrategyFilters ],
+    );
+
+    const updateStrategyFilters = useCallback(
+        (id: string, node: ICategoryStrategiesTypes) => {
+            const strategyFilterToFind = currentStrategyFilters.find(({ categoryId }) => categoryId === id);
+
+            (isNil(strategyFilterToFind)
```

Comment:
Maybe the naming here is not correct. This is an add or remove filter node method.